### PR TITLE
Download directories containing modules

### PR DESCRIPTION
**Pull request by jedesah**
- Add notion of `Node` in Filesystem to convey more information than simply `PathSegment`
- Simplify a few tests I happened to touch to use some new methods (saveThese)
- When downloading a directory, ignore any `Function` files so as to not throw an exception attempting to read from such files

### CODE REVIEW

**Comment by jedesah** [comment]
So what are we doing about Appveyor build failures?

File: connector/src/main/scala/quasar/connector/BackendModule.scala [hunk 1]
```diff
@@ -110,7 +110,8 @@ trait BackendModule {
           explanation <- QueryFileModule.explain(pp.repr)
         } yield ExecutionPlan(Type, explanation, pp.paths)).run.run
 
-      case QueryFile.ListContents(dir) => QueryFileModule.listContents(dir).run.value
+      case QueryFile.ListContents(dir) =>
+        QueryFileModule.listContents(dir).run.value.map(_.map(set => set.map(Node.fromSegment)))
```
Comment (jsantos17) [inline connector/src/main/scala/quasar/connector/BackendModule.scala]: I think it'd be slightly cleaner if you mapped `fromSegment` before running the transformer. Something like:

```
QueryFileModule.listContents(dir).map(set => set.map(Node.fromSegment)).run.value
```

File: core/src/main/scala/quasar/fs/mount/Mounting.scala [hunk 2]
```diff
@@ -112,10 +112,18 @@ object Mounting {
       lookupConfig(path).flatMap(config =>
         EitherT.right(OptionT(viewConfig.getOption(config).map(ViewConfig.tupled).point[FreeS])))
 
+    def lookupViewConfigIgnoreError(path: AFile): OptionT[FreeS, ViewConfig] =
+      lookupViewConfig(path).run
```
Comment (jsantos17) [inline core/src/main/scala/quasar/fs/mount/Mounting.scala]: I believe you can replace this with `OptionT(lookupViewConfig(path).toOption.run.run.map(_.join))`. Would be even nicer if we had `squash`. It'd be `lookupViewConfig(path).toOption.squash`.

Just a suggestion, change it if you want.

File: connector/src/main/scala/quasar/connector/BackendModule.scala (hunk 1 above)
Comment (jedesah) [reply connector/src/main/scala/quasar/connector/BackendModule.scala]: Agreed! Thanks for pointing that out!

File: connector/src/main/scala/quasar/fs/package.scala [hunk 3]
```diff
@@ -37,6 +39,52 @@ package object fs extends PhysicalErrorPrisms {
     def apply[F[_]](implicit F: MonadFsErr[F]): MonadFsErr[F] = F
   }
 
+  trait Node {
```
[the remark this answers is not in the document]
Comment (jedesah) [reply connector/src/main/scala/quasar/fs/package.scala]: Well, my thinking was that a connector can add an additional `Node` type in a typed manner. I don't know if that's worth the loss of exhaustivity checking though... What do you think?

File: core/src/main/scala/quasar/fs/mount/Mounting.scala (hunk 2 above)
Comment (jedesah) [reply core/src/main/scala/quasar/fs/mount/Mounting.scala]: Added `squash`. It sucks that (as I far as I can reason) it's not easy to make it generic in Scala but seems like it's worth defining it specifically for `OptionT`. Although I didn't look for them, I wouldn't be surprised if there were other parts of the codebase that could make use of `squash`

**Comment by jsantos17** [comment]
@jedesah Approved. I'll merge this after conflicts are resolved.

**Pull request merged**